src/util-decode-mime.c

```diff
@@ -1844,8 +1845,11 @@ static uint8_t * FindMimeHeaderToken(MimeDecField *field, const char *search_sta
     fptr = FindBuffer(field->value, field->value_len, (const uint8_t *)search_start, strlen(search_start));
     if (fptr != NULL) {
         fptr += strlen(search_start); /* Start at end of start string */
-        tok = GetToken(fptr, field->value_len - (fptr - field->value), search_end,
-                &tptr, tlen);
+        uint32_t search_len = field->value_len - (fptr - field->value);
+        if (max_len) {
```

Comment:
if `max_len` is set and the field value is bigger than it, we should set an event that we can match on from the rules.

Reply:
Will do.